### ISSUE TITLE
Consider making a napari frontend

### ISSUE THREAD

**Issue by niksirbi** [neuroinformatics-unit/movement#31]
`napari` has [multiple built-in layer types](https://napari.org/stable/guides/layers.html) which could be useful for `movement`.

- The `Image` layer could be used to display video frames
- The `Points` layer can be used to represent keypoints, see [this example](https://napari.org/stable/tutorials/annotation/annotate_points.html)
- The `Tracks` layer can be used to visualise keypoint trajectories across time
- The `Vectors` layer can be used to show stuff like head orientation
- The `Shapes` layer can be used to create and show ROIs

Napari widgets can be created easily.

**Comment by niksirbi** [comment]
Work in progress in the [napari branch](https://github.com/neuroinformatics-unit/movement/tree/napari).

**Comment by niksirbi** [comment]
Useful resources for designing napari plugins:
- [napari plugin desing workshop slides](https://docs.google.com/presentation/d/1Y237VmuhYwwtDFCA2Pteb_yAFiSJPN7hTUyq3nh7RX8/edit#slide=id.p)
- [list o popular napari plugins to try out](https://docs.google.com/spreadsheets/d/1ns8ZtkQeOp82TNpu2zbQCQu6KWExiyyLKijdSvZF-G8/edit#gid=0)

**Comment by niksirbi** [comment]
The current work-in progress is on this [branch](https://github.com/neuroinformatics-unit/movement/tree/napari-pose-reader).

**Comment by niksirbi** [comment]
While working on #112,  I realised that it was all getting too complex for one pull request, so I'm shifting strategy.

We decided to proceed as following:
- Make a [napari-dev](https://github.com/neuroinformatics-unit/movement/tree/napari-dev) branch to act as a temporary "main" branch for the development of the napari plugin + widget
- Break down the content of this PR into smaller, more manageable chunks and open them as individual PRs against `napari-dev`.

Once this process is well underway, and `napari-dev` catches up with what's here, I'll close PR #112 and delete the `napari-pose-reader` branch.

**Comment by sfmig** [comment]
I was just chatting to Sanna and we remembered she left valuable feedback re ROIs in the WAZP repo: I link it [here](https://github.com/SainsburyWellcomeCentre/WAZP/issues/98) in case it is useful

**Comment by niksirbi** [comment]
Yeah, and I implemented those suggestions in a [PR](https://github.com/SainsburyWellcomeCentre/WAZP/pull/103) I never merged.
